rewrite-java/src/main/java/org/openrewrite/java/search/SemanticallyEqual.java

```diff
@@ -313,15 +322,9 @@ public J.Case visitCase(J.Case _case, J j) {
                 }
 
                 J.Case compareTo = (J.Case) j;
-                if (_case.getStatements().size() != compareTo.getStatements().size()) {
-                    isEqual.set(false);
-                    return _case;
-                }
-
-                this.visit(_case.getPattern(), compareTo.getPattern());
-                for (int i = 0; i < _case.getStatements().size(); i++) {
-                    this.visit(_case.getStatements().get(i), compareTo.getStatements().get(i));
-                }
+                this.visitList(_case.getStatements(), compareTo.getStatements());
+                this.visit(_case.getBody(), compareTo.getBody());
+                this.visitList(_case.getExpressions(), compareTo.getExpressions());
```

Comment:
The comparison of `body` was missing and `pattern` has been deprecated in favor of `expressions`.